src/transformation/visitors/expression-list.ts

```diff
@@ -5,12 +5,13 @@ import { TransformationContext, tempSymbolId } from "../context";
 import { LuaLibFeature, transformLuaLibFunction } from "../utils/lualib";
 import { transformInPrecedingStatementScope } from "../utils/preceding-statements";
 import { isConstIdentifier } from "../utils/typescript";
+import { isOptionalContinuation } from "./optional-chaining";
 
 export function shouldMoveToTemp(context: TransformationContext, expression: lua.Expression, tsOriginal?: ts.Node) {
     return (
         !lua.isLiteral(expression) &&
         !(lua.isIdentifier(expression) && expression.symbolId === tempSymbolId) && // Treat generated temps as consts
-        !(tsOriginal && isConstIdentifier(context, tsOriginal))
+        !(tsOriginal && (isConstIdentifier(context, tsOriginal) || isOptionalContinuation(tsOriginal)))
```

Comment:
What is the purpose of this?

Reply:
If an identifier is an optional continuation (ts temp variable), it doesn't need not be moved to temp. Noticed this when debugging the newly added tests.

Reply:
ah, ok